### PR TITLE
Add label and switch ocdialog close button to button element

### PR DESCRIPTION
**Pull request by Pytal**
- For https://github.com/nextcloud/server/issues/36715

## Summary

Before | After
--- | ---
image | image

Without a label screenreader users cannot identify the interactive element

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by szaimen — APPROVED** [review]
LGTM but didnt test

File: core/src/jquery/css/jquery.ocdialog.scss [hunk 1]
```diff
@@ -50,8 +50,10 @@
 
 .oc-dialog-close {
 	position: absolute;
-	top: 0;
-	right: 0;
+	width: 44px !important;
+	height: 44px !important;
+	top: 5px;
```
Comment (artonge) [inline core/src/jquery/css/jquery.ocdialog.scss:55]: I think we are supposed to use multiple of 4 for spacing. Is there a rationale behind 5?

Comment (Pytal) [reply core/src/jquery/css/jquery.ocdialog.scss]: Nope :) switched to `4px`

File: core/src/jquery/ocdialog.js [hunk 2]
```diff
@@ -177,7 +177,7 @@ $.widget('oc.ocdialog', {
 			break
 		case 'closeButton':
 			if (value) {
-				const $closeButton = $('<a class="oc-dialog-close" tabindex="0"></a>')
+				const $closeButton = $(`<button class="oc-dialog-close" aria-label="${t('core', 'Close')}"></button>`)
```
Comment (artonge) [inline core/src/jquery/ocdialog.js:180]: 'Close dialog' ?

**Review by artonge — APPROVED** [review]

**Review by JuliaKirschenheuter — CHANGES_REQUESTED** [review]
please see the comment from @artonge

**Review by JuliaKirschenheuter — APPROVED** [review]

**Pull request merged**